### PR TITLE
Rgaa card favorite tooltip

### PR DESCRIPTION
**Pull request by StefanoA1**
**Detailed purpose of the PR**

- fixes battle opponent list tooltip issue
- adds ReactTooltip for Favorite (for Catalog's Cards)

**Result and observation**
Keyboard:
Screenshot 2023-03-09 at 21 12 56
Mouse:
Screenshot 2023-03-09 at 21 13 05
Keyboard:
Screenshot 2023-03-09 at 21 15 30
Mouse:
Screenshot 2023-03-09 at 21 15 36

**Testing Strategy**

- [x] Already covered by tests
- [x] Manual testing
- [x] Unit testing

### CODE REVIEW

**Comment by StefanoA1** [comment]
[12.11 - Dans chaque page web, les contenus additionnels apparaissant au survol, à la prise de focus ou à l'activation d'un composant d'interface sont-ils, si nécessaire, atteignables au clavier ?](https://trello.com/c/RBLiiHXe/97-1211-dans-chaque-page-web-les-contenus-additionnels-apparaissant-au-survol-a%CC%80-la-prise-de-focus-ou-a%CC%80-lactivation-dun-composant)

File: packages/@coorpacademy-components/src/molecule/card/favorite.css [hunk 1]
```diff
@@ -47,48 +45,30 @@
...
-.showToolTip::after {
-  content: "";
-  position: absolute;
-  top: 8px;
-  left: 100%;
-  border: solid 5px ;
-  border-color: transparent transparent transparent color(black a(80%));
+.favoriteWrapper .favorite + [class*="__react_component_tooltip"]:not([class*="toolTipReact"]) {
+  position: sticky;
+  top: unset !important;
```
Comment (silouone) [inline packages/@coorpacademy-components/src/molecule/card/favorite.css:55]: 🤔

File: packages/@coorpacademy-components/src/molecule/card/favorite.css [hunk 2]
```diff
@@ -47,48 +45,30 @@
...
+  left: unset !important;
```
Comment (silouone) [inline packages/@coorpacademy-components/src/molecule/card/favorite.css:56]: 🤔

File: packages/@coorpacademy-components/src/molecule/card/favorite.css (hunk 1 above)
Comment (StefanoA1) [reply packages/@coorpacademy-components/src/molecule/card/favorite.css:55]: -> `!important` + selectors specificity to override css props used with !important inside the external tooltip lib that we use (no other way to override it)

File: packages/@coorpacademy-components/src/atom/battle-opponent/style.css [hunk 3]
```diff
@@ -112,6 +112,18 @@
...
+.toolTipWrapper {
+  position: absolute;
+  width: 100%;
+}
+
+.card .tooltipIconContainer + .toolTipWrapper [class*="__react_component_tooltip"]:not([class*="toolTipReact"]) {
+  position: sticky;
+  top: unset !important;
+  left: unset !important;
+  max-width: 75%;
```
Comment (StefanoA1) [inline packages/@coorpacademy-components/src/atom/battle-opponent/style.css:124]: to fix external tooltip lib's issue of keyboard navigation positioning on key only events

**Review by DjamelSoualmi — CHANGES_REQUESTED** [review]

File: packages/@coorpacademy-components/src/molecule/card/favorite.js [hunk 4]
```diff
@@ -35,46 +38,113 @@ const Favorite = (
...
-        ) : (
-          <MoreIcon
-            className={style.moreIcon}
-            style={{color: darkColor}}
-            width={13}
-            height={13}
-            aria-label={ariaLabel.addToFavorite}
-          />
+        {icon}
+      </button>
```
Comment (DjamelSoualmi) [inline packages/@coorpacademy-components/src/molecule/card/favorite.js:137]: si tu peux utiliser Button (lle composant existant ) ça serait cool

**Review by DjamelSoualmi — CHANGES_REQUESTED** [review]

**Review by antonia2929 — APPROVED** [review]

File: packages/@coorpacademy-components/src/molecule/card/favorite.js (hunk 4 above)
Comment (StefanoA1) [reply packages/@coorpacademy-components/src/molecule/card/favorite.js:137]: le pb c'est que `input` ne supporte pas les children

**Review by DjamelSoualmi — APPROVED** [review]

**Pull request merged**